Unit tests in tests/slsa_analyzer/build_tool/test_gradle.py rely on network

This applies for unit tests in `tests/slsa_analyzer/build_tool/test_gradle.py` that test the method `get_group_ids` https://github.com/oracle/macaron/blob/5522ec103d586789ec1a99db29c02d94a7949656/src/macaron/slsa_analyzer/build_tool/gradle.py#L170

This method eventually invokes `get_group_id` method that creates a subprocess to call `gradlew` shipped with Macaron or the `gradlew` within the target repository source code. https://github.com/oracle/macaron/blob/5522ec103d586789ec1a99db29c02d94a7949656/src/macaron/slsa_analyzer/build_tool/gradle.py#L231-L237

`gradlew` however, on first run would try to download the gradle distribution from the internet to bootstrap it. If it cannot download the gradle distribution, the relevant unit tests will fail. 

To reproduce:
```
# Run within the Macaron repository. Make sure to turn off your internet connection.
mv ~/.gradle ~/.gradle_bak
make setup
pytest tests/slsa_analyzer/build_tool/test_gradle.py
```

The error:
```
FAILED tests/slsa_analyzer/build_tool/test_gradle.py::test_get_group_ids_nested_projects - AssertionError: assert set() == {'io.micronaut'}
  
  Extra items in the right set:
  'io.micronaut'
  
  Full diff:
  + set()
  - {
  -     'io.micronaut',
  - }
FAILED tests/slsa_analyzer/build_tool/test_gradle.py::test_get_group_ids_separate_projects - AssertionError: assert set() == {'io.micronaut.foo', 'io.micronaut.bar'}
  
  Extra items in the right set:
  'io.micronaut.foo'
  'io.micronaut.bar'
  
  Full diff:
  + set()
  - {
  -     'io.micronaut.bar',
  -     'io.micronaut.foo',
  - }
FAILED tests/slsa_analyzer/build_tool/test_gradle.py::test_get_group_ids_timeout[invalid-expected1] - AssertionError: assert set() == {'io.micronaut'}
  
  Extra items in the right set:
  'io.micronaut'
  
  Full diff:
  + set()
  - {
  -     'io.micronaut',
  - }
==================== 3 failed, 26 passed in 42.13s ====================
```

I wonder if we should look into fixing it somehow? Or it's okay to leave it as it is.

--- a/src/macaron/slsa_analyzer/build_tool/gradle.py
+++ b/src/macaron/slsa_analyzer/build_tool/gradle.py
@@ -167,42 +167,6 @@ class Gradle(BaseBuildTool):
         # We use Macaron's built-in gradlew as a fallback option.
         return os.path.join(os.path.join(macaron.MACARON_PATH, "resources"), "gradlew")
 
-    def get_group_ids(self, repo_path: str) -> set[str]:
-        """Get the group ids of all Gradle projects in a repository.
-
-        A Gradle project is a directory containing a ``build.gradle`` file.
-        According to the Gradle's documentation, there is a one-to-one mapping between
-        a "project" and a ``build.gradle`` file.
-        See: https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html.
-
-        Note: This method makes the assumption that projects nested in a parent project
-        directory has the same group id with the parent. This behavior is consistent with
-        the behavior of the ``get_build_dirs`` method.
-
-        Parameters
-        ----------
-        repo_path: str
-            The absolute path to a repository containing Gradle projects.
-
-        Returns
-        -------
-        set[str]
-            The set of group ids of all Gradle projects in the repository.
-        """
-        gradle_exec = self.get_gradle_exec(repo_path)
-        group_ids = set()
-
-        for gradle_project_relpath in self.get_build_dirs(repo_path):
-            gradle_project_path = os.path.join(repo_path, gradle_project_relpath)
-            group_id = self.get_group_id(
-                gradle_exec=gradle_exec,
-                project_path=gradle_project_path,
-            )
-            if group_id:
-                group_ids.add(group_id)
-
-        return group_ids
-
     def get_group_id(self, gradle_exec: str, project_path: str) -> str | None:
         """Get the group id of a Gradle project.
 

--- a/tests/slsa_analyzer/build_tool/test_gradle.py
+++ b/tests/slsa_analyzer/build_tool/test_gradle.py
@@ -8,7 +8,6 @@ from pathlib import Path
 import pytest
 
 from macaron.code_analyzer.call_graph import BaseNode
-from macaron.config.defaults import load_defaults
 from macaron.slsa_analyzer.build_tool.base_build_tool import BuildToolCommand
 from macaron.slsa_analyzer.build_tool.gradle import Gradle
 from macaron.slsa_analyzer.build_tool.language import BuildLanguage
@@ -41,81 +40,6 @@ def test_gradle_build_tool(gradle_tool: Gradle, macaron_path: str, mock_repo: st
     base_dir = Path(__file__).parent
     ctx = prepare_repo_for_testing(mock_repo, macaron_path, base_dir)
     assert gradle_tool.is_detected(ctx.component.repository.fs_path) == expected_value
-
-
-def test_get_group_ids_nested_projects(tmp_path: Path, gradle_tool: Gradle) -> None:
-    """Test the ``get_group_ids`` method in case there are Gradle projects nested under a root project.
-
-    In this case, we should only obtain the group id of the root project, making the assumption
-    that all subprojects under it have the same group id.
-
-    This is consistent with the behavior of the ``get_build_dirs`` method.
-    """
-    repo_dir = tmp_path.joinpath("repo")
-    subproject_a_dir = repo_dir.joinpath("subprojecta")
-    subproject_b_dir = repo_dir.joinpath("subprojectb")
-
-    subproject_a_dir.mkdir(parents=True)
-    subproject_b_dir.mkdir(parents=True)
-
-    with open(repo_dir.joinpath("build.gradle"), "w", encoding="utf-8") as file:
-        file.write('group = "io.micronaut"')
-    with open(subproject_a_dir.joinpath("build.gradle"), "w", encoding="utf-8") as file:
-        file.write('group = "io.micronaut.foo"')
-    with open(subproject_b_dir.joinpath("build.gradle"), "w", encoding="utf-8") as file:
-        file.write('group = "io.micronaut.bar"')
-
-    assert set(gradle_tool.get_group_ids(str(repo_dir))) == {"io.micronaut"}
-
-
-def test_get_group_ids_separate_projects(tmp_path: Path, gradle_tool: Gradle) -> None:
-    """Test the ``get_group_ids`` method in case there are multiple separate Gradle projects in a repo.
-
-    "Separate projects" means they are in different directories in the repo.
-    """
-    repo_dir = tmp_path.joinpath("repo")
-
-    project_a_dir = repo_dir.joinpath("subprojecta")
-    project_b_dir = repo_dir.joinpath("subprojectb")
-
-    project_a_dir.mkdir(parents=True)
-    project_b_dir.mkdir(parents=True)
-
-    with open(project_a_dir.joinpath("build.gradle"), "w", encoding="utf-8") as file:
-        file.write('group = "io.micronaut.foo"')
-    with open(project_b_dir.joinpath("build.gradle"), "w", encoding="utf-8") as file:
-        file.write('group = "io.micronaut.bar"')
-
-    assert set(gradle_tool.get_group_ids(str(repo_dir))) == {
-        "io.micronaut.foo",
-        "io.micronaut.bar",
-    }
-
-
-@pytest.mark.parametrize(("timeout", "expected"), [("0", set()), ("invalid", {"io.micronaut"})])
-def test_get_group_ids_timeout(tmp_path: Path, gradle_tool: Gradle, timeout: str, expected: set) -> None:
-    """Test the timeout configuration on ``get_group_ids`` method."""
-    repo_dir = tmp_path.joinpath("repo")
-    repo_dir.mkdir()
-
-    with open(repo_dir.joinpath("build.gradle"), "w", encoding="utf-8") as file:
-        file.write('group = "io.micronaut"')
-
-    user_config_path = str(tmp_path.joinpath("config.ini"))
-    user_config_input = f"""
-        [builder.gradle.runtime]
-        build_timeout = {timeout}
-    """
-    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
-        user_config_file.write(user_config_input)
-
-    # We don't have to worry about modifying the ``defaults`` object causing test
-    # pollution here, since we reload the ``defaults`` object before every test with the
-    # ``setup_test`` fixture.
-    load_defaults(user_config_path)
-    gradle_tool.load_defaults()
-
-    assert set(gradle_tool.get_group_ids(str(repo_dir))) == expected
 
 
 @pytest.mark.parametrize(

